### DISCUSSION TITLE
Word vectors - which algorithm is used

### DISCUSSION THREAD

**Discussion by Joozty** [explosion/spaCy#9558] [Help: Other Questions] [answered]
Models like `en_core_web_md` are shipped with pretrianed word embeddings. I wonder what algorithm did you use to train these vectors?  Is it [Tomas Mikolov’s original Word2vec implementation](https://arxiv.org/abs/1301.3781)?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @Joozty , this is documented further in the  [model docs](https://spacy.io/models/en#en_core_web_md)! Can take a look there! :+1:

**Reply by Joozty** [reply]
I am afraid it's not there. I already have a look before writing this question. Maybe I am overlooking something? 🙂

**Reply by ljvmiranda921** [reply]
We use pretrained vectors and did not train our own. In the [model docs](https://spacy.io/models/en#en_core_web_md), we're using the [Glove CommonCrawl](https://nlp.stanford.edu/projects/glove/). Hope it's clearer 👍

**Reply by adrianeboyd** [reply]
Also note that the vectors in the current English pipelines have been passed down through various spacy versions over the years and have ended up with some quirks, mainly related to case sensitivity, and they no longer correspond exactly to any provided glove vectors. We're hoping to replace them with our own vectors in the future.
